src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()) {
+            storage.save(addressBook);
+        }
         return result;
```

Comment:
The header comment of this method should have been updated to correctly reflect the changes in the behaviour of the code.